py/selenium/webdriver/common/actions/wheel_input.py

```diff
@@ -21,6 +21,34 @@
 from selenium.webdriver.remote.webelement import WebElement
 
 
+class ScrollOrigin:
+
+    def __init__(self, origin: Union[str, WebElement], x_offset: int, y_offset: int) -> None:
+        self._origin = origin
+        self._x_offset = x_offset
+        self._y_offset = y_offset
+
+    @classmethod
+    def from_element(cls, element: WebElement, x_offset: int = 0, y_offset: int = 0):
+        return cls(element, x_offset, y_offset)
+
+    @classmethod
+    def from_viewport(cls, x_offset: int = 0, y_offset: int = 0):
+        return cls('viewport', x_offset, y_offset)
+
+    @property
+    def origin(self) -> Union[str, WebElement]:
```

Comment:
read-only ok for these? (seems ok to me)

Reply:
Yes, I thought you had to explicitly add a setter to make it not read only?

Reply:
Yep I'm just wondering if that was the intent (vs allowing them to be set and modified afterwards on the instance)

Reply:
Ideally the constructor would also be "private" like we can do it in Java. Ruby we are marking the constructor `@api private` to annotate that. Not sure if there is an equivalent in Python. The ideal is to only create the instance via class method, which properly sets appropriate values/defaults.